src/autobridgecalculator.js

```diff
@@ -70,14 +73,17 @@ AutobridgeCalculator.prototype.calculate = function() {
       continue;
     }
 
-    if (legOneTakerGetsFunded.compareTo(legTwoTakerPaysFunded) > 0) {
+    //  using private fields for speed
```

Comment:
So in the compareTo function, you could skip the from_json call if it is already an Amount.

The general point is that there is sometimes a tradeoff between code quality and performance, and we have to draw a line somewhere so that we don't let perf optimizations create too much complexity. Ideally javascript would have enforced private methods (like many other languages) so we wouldn't even have to discuss it, but I think we should just consider such tricks as off-limits and find other workarounds, which always exist.